packages/db/src/meta/pouch/adapters/fs.ts

```diff
@@ -15,10 +15,25 @@ import Config from "@truffle/config";
 export interface DatabasesSettings {
   directory: string;
 }
+type UserConfigDbSettings = {
+  saveLocally: boolean | undefined,
+};
 
-export const getDefaultSettings: GetDefaultSettings = () => ({
-  directory: path.join(Config.getTruffleDataDirectory(), ".db", "json")
-});
+type UserConfig = {
+  get: (key: "db") => UserConfigDbSettings,
+};
+
+export const getDefaultSettings: GetDefaultSettings = () => {
+  const userConfig: UserConfig = Config.getUserConfig();
+  
+  return {
+    directory: path.join(
+      userConfig.get("db")?.saveLocally 
+        ? new Config().working_directory  
```

Comment:
`new Config()` is generally considered private, and you should do `Config.detect()` or `Config.default().merge({ ... })` instead.

Earlier I told you that we should just do `Config.default()`, but I think that's wrong: I think we do need a proper `Config.detect()` here

Reply:
:+1:  Switched back to `.detect()`